components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/VersionsControllerTest.java

```diff
@@ -0,0 +1,266 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import net.sf.json.JSONObject;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for the {@link VersionsController} Component,
+ * only the overridden methods from {@link AbstractSimpleController} are tested here
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ ComponentManagerRegistry.class })
+public class VersionsControllerTest
+{
+
+    @Rule
+    public MockitoComponentMockingRule<PatientDataController<String>> mocker =
+        new MockitoComponentMockingRule<PatientDataController<String>>(VersionsController.class);
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private XWikiDocument doc;
+
+    @Mock
+    private BaseObject firstOntologyVersion;
+
+    @Mock
+    private BaseObject secondOntologyVersion;
+
+    @Mock
+    private ComponentManager contextComponentManager;
+
+    @Mock
+    private DistributionManager distributionManager;
+
+    private static final EntityReference ONTOLOGY_VERSION_CLASS_REFERENCE =
+        new EntityReference("OntologyVersionClass", EntityType.DOCUMENT, Constants.CODE_SPACE_REFERENCE);
+
+    private static final String DATA_NAME = "versions";
+
+    private static final String PHENOTIPS_VERSION_STRING = "PHENOTIPS_VERSION";
+
+    @Before
+    public void setUp() throws Exception
+    {
+        MockitoAnnotations.initMocks(this);
+
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+
+        DocumentReference patientDocument = new DocumentReference("wiki", "patient", "00000001");
+        doReturn(patientDocument).when(this.patient).getDocument();
+        doReturn(this.doc).when(this.documentAccessBridge).getDocument(patientDocument);
+
+        doReturn("first").when(this.firstOntologyVersion).getStringValue("name");
+        doReturn("1.0").when(this.firstOntologyVersion).getStringValue("version");
+        doReturn("second").when(this.secondOntologyVersion).getStringValue("name");
+        doReturn("2.0").when(this.secondOntologyVersion).getStringValue("version");
+        List<BaseObject> ontologyVersionList = Arrays.asList(firstOntologyVersion, secondOntologyVersion);
+        doReturn(ontologyVersionList).when(this.doc).getXObjects(ONTOLOGY_VERSION_CLASS_REFERENCE);
+
+        PowerMockito.mockStatic(ComponentManagerRegistry.class);
```

Comment:
This is [usually done](https://github.com/phenotips/phenotips/blob/master/components/patient-data/impl/src/test/java/org/phenotips/data/internal/PhenoTipsDisorderTest.java#L66) without PowerMockito.